File: src/IrisManager.Domain/Entities/Appointment.cs
using System.ComponentModel.DataAnnotations;

namespace IrisManager.Domain.Entities
{
    public class Appointment
    {

        public int Id { get; set; }

        [Required]
        public int CustomerId { get; set; }
        public Customer? Customer { get; set; }

        [Required]
        public int StylistId { get; set; }
        public Stylist? Stylist { get; set; }

        [Required]
        public int ServiceId { get; set; }
        public Service? Service { get; set; }

        [Required]
        public DateTime StartTime { get; set; }

        [Required]
        public DateTime EndTime { get; set; }

        [Required]
        public string Status { get; set; } = "Scheduled";
    }
}

File: src/IrisManager.Client/src/app/core/models/appointment.ts
export interface Appointment {
  id: number;
  customerId: number;
  customerName: string;
  isCustomerActive?: boolean;
  stylistId: number;
  stylistName: string;
  serviceId: number;
  serviceName: string;
  servicePrice?: number;
  startTime: string;
  endTime: string;
  status: 'Scheduled' | 'Completed' | 'Cancelled' | string;
  paymentMethod?: string;
}

export interface AppointmentCreateDTO {
  customerId: number;
  stylistId: number;
  serviceId: number;
  startTime: string;
  status?: string;
  paymentMethod?: string;
}

export interface AppointmentUpdateDTO {
  id: number;
  customerId: number;
  stylistId: number;
  serviceId: number;
  startTime: string;
  status?: string;
  paymentMethod?: string;
}

export interface AppointmentUpdateStatusDTO {
  status: string;
}